opennms-dao/src/main/java/org/opennms/netmgt/dao/hibernate/EventConfEventDaoHibernate.java

```diff
@@ -75,6 +77,38 @@ public List<EventConfEvent> filterEventConf(final String uei, final String vendo
         return findWithPagination(queryBuilder.toString(), queryParamList.toArray(), offset, limit);
     }
 
+    @Override
+    public Map<String, Object> findBySourceId(Long sourceId, Integer totalRecords, Integer offset, Integer limit) {
+        int resultCount = (totalRecords != null) ? totalRecords : 0;
+        List<Object> queryParams = new ArrayList<>();
+        List<String> conditions = new ArrayList<>();
+
+        // Add filter conditions dynamically
+        if (sourceId != null && sourceId>0) {
+            conditions.add("e.source.id = ?");
+            queryParams.add(sourceId);
+        }
+
+        String whereClause = conditions.isEmpty() ? "" : " where " + String.join(" OR ", conditions);
```

Comment:
There is only condition, can we just to do 
`String whereClause = " where e.source.id = ?";`